addons/rating/models/rating.py

```diff
@@ -0,0 +1,146 @@
+    rating = fields.Float(string="Rating", group_operator="avg", default=-1, help="Rating value")
+    feedback = fields.Text('Feedback reason', help="Reason of the rating")
+    access_token = fields.Char(string='Security Token', default=new_access_token, help="Access token to set the rating of the value")
+
+    @api.model
+    def apply_rating(self, rate, res_model=None, res_id=None, token=None):
+        """ apply a rating for given res_model/res_id or token. If the res_model is a mail.thread
+            object, a message will be post in the chatter.
+            :param float rating : the rating value to apply
+            :param list res_id : id of the rated object.
+            :param string res_model : name of model.
+            :param string token : access token
+            :return the rating record
+        """
+        domain = [('access_token', '=', token)] if token else [('res_model', '=', res_model), ('res_id', '=', res_id)]
+        rating = self.search(domain, limit=1)
+        if rating:
+            rating.write({'rating' : rate})
+            if hasattr(self.env[rating.res_model], 'message_post'):
+                record = self.env[rating.res_model].sudo().browse(rating.res_id)
+                record.message_post(
+                    body="%s %s <br/><img src='rating/static/src/img/rating_%s.png' style='width:20px;height:20px'/>"
+                    % (rating.sudo().partner_id.name, _('rated it'), rate))
+        return rating
+
+    @api.multi
+    def reset(self):
+        for record in self:
+            record.write({
+                'rating': -1,
+                'access_token': record.new_access_token(),
+                'feedback' : False
+            })
+
+
+
+class RatingMixin(models.AbstractModel):
+
+    _name = 'rating.mixin'
+    _description = "Rating Mixin"
+
+    rating_ids = fields.One2many('rating.rating', 'res_id', string='Rating', domain=lambda self: [('res_model', '=', self._name)])
+
+    @api.multi
+    def rating_send_request(self, template, partner_id, rated_partner_id, reuse_rating=True):
+        """ This method create (empty) rating objects for the current recordsets and send this request by mail (given email
+            template) with the given rated_partner_id and partner_id as recipient and sender of the email.
+            :param email.template template : the email template to send. The res_model of the template must be 'rating.rating'.
+            :param string res_model : mddel name of the object to rated_partner_id
+            :param int res_id : id the record to rate
+            :param res.partner partner_id : the recipient partner
+            :param res.partner rated_partner_id : the sender partner
+            :param bool reuse_rating : if True, the rating of the current objects will be reset. Otherwise a new rating will be create
+        """
+        if rated_partner_id.email and partner_id.email:
```

Comment:
Just to avoid unnecessary indentation, I would write

if not rated_partner_id.email or not partner_id.email:
  return False
# remaining of the method
return True